roles/zuul/files/jobs/bonnyci.yaml

```diff
@@ -0,0 +1,43 @@
+- publisher:
+    name: bonnyci-logs
+    publishers:
+      - scp:
+          site: bonnyci-scp
+          files:
+            - target: 'logs/$LOG_PATH'
+              source: '/opt/bonnyci/logs/**'
+              keep-hierarchy: true
+              copy-after-failure: true
+
+- job-template:
+    name: 'bonnyci-run-{pipeline}'
+    pipeline: '{pipeline}'
+    test_log_dir: /opt/bonnyci/logs
+    builders:
+      - zuul-git-prep
+      - shell: |
+            #!/bin/bash
+            if [[ ! -f .bonnyci/run.sh ]]; then
+                echo "ERROR: No test script found at .bonnyci/run.sh, skipping"
+                exit 1
+            elif [[ ! -x .bonnyci/run.sh ]]; then
+                echo "ERROR: Test script found at .bonnyci/run.sh but it is not executable"
+                exit 1
+            fi
+
+            export BONNYCI_TEST_PIPELINE={pipeline}
+            export BONNYCI_TEST_LOG_DIR={test_log_dir}
```

Comment:
If we set these in bonnyci_functions.py then we can use BONNYCI_TEST_LOG_DIR in the publisher source and won't need to pass in a parameter to set it, reducing the number of places this is hard-coded and making it easier to update. But since this is only meant to be a temporary solution I don't feel very strongly about making it easy to update.